shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -29,25 +63,154 @@ - (void)touchesBegan:(NSSet*)touches withEvent:(UIEvent*)event {
   }
 }
 
-#if FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
 - (void)applicationDidEnterBackground:(UIApplication *)application {
+  #if FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
   // The following keeps the Flutter session alive when the device screen locks
-  // in debug mode. It allows continued use of features like hot reload and 
+  // in debug mode. It allows continued use of features like hot reload and
   // taking screenshots once the device unlocks again.
   //
-  // Note the name is not an identifier and multiple instances can exist. 
+  // Note the name is not an identifier and multiple instances can exist.
   _debugBackgroundTask = [application beginBackgroundTaskWithName:@"Flutter debug task"
                                                 expirationHandler:^{
       FTL_LOG(WARNING) << "\nThe OS has terminated the Flutter debug connection for being "
                           "inactive in the background for too long.\n\n"
                           "There are no errors with your Flutter application.\n\n"
                           "To reconnect, launch your application again via 'flutter run";
       }];
+  #endif  // FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
+  for (id<FlutterPlugin> plugin in _pluginDelegates) {
+    if ([plugin respondsToSelector:_cmd]) {
+      [plugin applicationWillEnterForeground:application];
```

Comment:
Fixed the bug, thanks!. I did not yet introduce the suggested macro or utility method because (per question above) we may want to simplify the signatures, eliding the `UIApplication` argument.